store/storetest/channel_store.go

```diff
@@ -304,10 +304,6 @@ func testChannelStoreUpdate(t *testing.T, ss store.Store) {
 	_, err = ss.Channel().Update(&o1)
 	require.NotNil(t, err, "Update should have failed because of missing key")
 
-	o1.Id = model.NewId()
-	_, err = ss.Channel().Update(&o1)
-	require.NotNil(t, err, "update should have failed because id change")
-
```

Comment:
This test has been there for a long time, through many refactors. How does it pass?

Reply:
It passes because it throws an error if no rows are updated.